From 9025fa8042729e34b3c02792961e774992096c41 Mon Sep 17 00:00:00 2001
From: arathinx <ayyappanx.rathinam@intel.com>
Date: Thu, 18 Jun 2020 13:36:07 +0800
Subject: [PATCH] DRVLIB_SW-3672: Resolve unreachable code warning in XPCS
 driver

pdata->mpllb should be always zero for 10G-xaui mode
---
 drivers/net/ethernet/lantiq/xpcs/xpcs.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/xpcs/xpcs.c b/drivers/net/ethernet/lantiq/xpcs/xpcs.c
index d2c9a4441864..aeac5a6573db 100644
--- a/drivers/net/ethernet/lantiq/xpcs/xpcs.c
+++ b/drivers/net/ethernet/lantiq/xpcs/xpcs.c
@@ -658,13 +658,8 @@ static int xpcs_teng_xaui_mode(struct xpcs_prv_data *pdata)
 	 * When this bit is set, PHY selects MPLLB to generate
 	 * Tx analog clocks on lane 0
 	 */
-	if (pdata->mpllb) {
-		XPCS_RGWR_VAL(pdata, PMA_MPLL_CMN_CTRL, MPLLB_SEL_3_1, 0);
-		XPCS_RGWR_VAL(pdata, PMA_MPLL_CMN_CTRL, MPLLB_SEL_0, 1);
-	} else {
-		XPCS_RGWR_VAL(pdata, PMA_MPLL_CMN_CTRL, MPLLB_SEL_3_1, 0);
-		XPCS_RGWR_VAL(pdata, PMA_MPLL_CMN_CTRL, MPLLB_SEL_0, 0);
-	}
+	XPCS_RGWR_VAL(pdata, PMA_MPLL_CMN_CTRL, MPLLB_SEL_3_1, 0);
+	XPCS_RGWR_VAL(pdata, PMA_MPLL_CMN_CTRL, MPLLB_SEL_0, 0);
 
 	xpcs_cfg_table(pdata);
 
